dashboard/app/models/section_based_experiment.rb

```diff
@@ -0,0 +1,37 @@
+# == Schema Information
+#
+# Table name: experiments
+#
+#  id                     :integer          not null, primary key
+#  name                   :string(255)
+#  type                   :string(255)
+#  start_time             :datetime
+#  end_time               :datetime
+#  created_at             :datetime         not null
+#  updated_at             :datetime         not null
+#  percentage             :integer
+#  earliest_section_start :datetime
+#  latest_section_start   :datetime
+#  section_id             :integer
+#
+
+class SectionBasedExperiment < Experiment
+  validates :percentage, inclusion: 0..100
+
+  def self.get_enabled(user: nil, section: nil)
+    return Experiment.none unless section
+    Experiment.where(type: SectionBasedExperiment.to_s).
+      where('percentage > ?', section.id % 100).
```

Comment:
Here and below, add a hash of the experiment name to section.id so that we don't always put the same sections/users into experiments.